crates/services/executor/src/executor.rs

```diff
@@ -549,14 +566,26 @@ where
             Ok(())
         };
 
-        while iter.peek().is_some() {
-            for transaction in iter {
-                execute_transaction(&mut *execution_data, transaction)?;
+        let relayed_tx_iter = forced_transactions.into_iter();
+        for transaction in relayed_tx_iter {
+            const RELAYED_GAS_PRICE: Word = 0;
+            execute_transaction(&mut *execution_data, transaction, RELAYED_GAS_PRICE)?;
```

Comment:
What do you think about handling error and emit the event here? It will remove `let mut relayed_tx: Option<RelayedTransactionId>` hack=) Plus maybe we can remove `ExecutorError::RelayedTransactionFailed` variant

Reply:
Hmm, and it will remove `MaybeCheckedTransaction::RelayedCheckedTransaction`!=)